### DISCUSSION TITLE
Is it possible to ‘copy’ a sound in realtime, whilst keeping copies in-sync?

### DISCUSSION THREAD

**Discussion by sbauly** [alnitak/flutter_soloud#157] [Q&A] [answered]
I am wondering if it is possible to play a sound, then copy it many times, with the copied sounds all playing at the same position as the original audio.

For example:

1. Load and play a sound.
2. Copy this sound multiple times, with each copy all playing at the same time & position as the original.

This would be useful, as each copied sound could have a different balance/pan, 3D position using the `play3d()` feature, or different filters applied to them.

Is this possible to do in realtime?

I believe this would require using the same `AudioSource` and creating new `SoundHandle`s each time the sound is copied - but I am not sure how to reliably keep the copied sounds in-sync with the original sound’s position in realtime.

Thanks!

**Comment by alnitak** [comment, marked as answer]
I think it's possible. You can play as many sound instances as you like in a pause state, then unpause them all together. Or if you want to play another instance while the other are playing, you can read the position and then seek the new one for that position. 
If you can I think it would be better to pause them all, read the position, seek the new handle and then play all to avoid possible lags between reading the position and the play of the new handle.

Remember that playing the same sound more than once implies that the amplitude of the resulting waveform can potentially double. This doubling of amplitude results in a louder sound and may be distorted even if it is not guaranteed that the sounds are perfectly aligned.

**Reply by sbauly** [reply]
Great, thanks! 

I think you're right, it's probably better to pause sounds before duplicating them, to keep them accurately in-sync. It's a little tricky to do in realtime.

I see a PR for new compressor and limiter filters, hopefully they should help with the increased amplitude caused by duplicating sounds :)

**Reply by alnitak** [reply]
> I see a PR for new compressor and limiter filters, hopefully they should help with the increased amplitude caused by duplicating sounds :)

Yes, exactly! Those new filters come up for this *sound overlapping*! :)
